lib/cdo/sauce_connect.rb

```diff
@@ -0,0 +1,113 @@
+require 'open3'
+require 'timeout'
+
+# Starts the Sauce Connect Proxy
+
+module Cdo
+  module SauceConnect
+    # How many seconds to wait for the "you may start your tests" message
+    SC_START_TIMEOUT_S = 120
+    SC_START_MESSAGE = "you may start your tests"
+    SC_STDOUT_PREFIX = "Sauce Connect Proxy"
+
+    class << self
+      # Starts the Sauce Connect Proxy, which allows tunneling connections from our local server to Sauce Labs
+      # This method blocks until sc prints the "you may start your tests" message to log/sc.log
+      #
+      # @param [Boolean] daemonize - if true, sc will continue running in the bg even when ruby exits
+      def start_sauce_connect(daemonize: false)
+        log_file = open_log_file
+
+        # Verify that required parameters are set in locals.yml
+        raise "saucelabs_username must be set in locals.yml" unless CDO.saucelabs_username
+        raise "saucelabs_authkey must be set in locals.yml, use 'Access Key' from https://app.saucelabs.com/user-settings" unless CDO.saucelabs_authkey
+        raise "saucelabs_tunnel_name must be set in locals.yml, 'cdo-tunnel' is a good default choice" unless CDO.saucelabs_tunnel_name
+
+        # Regexes defining the localhost domains we want to tunnel:
+        tunnel_domains = [
+          %q(.*\.code.org),
+          %q(.*\.csedweek.org),
+          %q(.*\.hourofcode.com),
+          %q(.*\.codeprojects.org),
+        ].join(',')
+
+        cmd = [
+          "sc", "run",
+          "--region", "us-west-1",
+          "--tunnel-domains", tunnel_domains,
+          "--tunnel-name", CDO.saucelabs_tunnel_name,
+        ]
+
+        env = {
+          "SAUCE_USERNAME" => CDO.saucelabs_username,
+          "SAUCE_ACCESS_KEY" => CDO.saucelabs_authkey
+        }
+
+        pid = Process.spawn(env, *cmd, out: log_file.path, err: log_file.path)
+        log "starting sc, pid = #{pid}, see log at #{log_file.path}"
+
+        # Block waiting for `sc` to print "you may start your tests"
+        tests_started, log_lines = tests_started?(log_file, pid)
+
+        if tests_started
+          log "success, sc is running"
```

Comment:
Worth surfacing some information here about whether we're daemonized or not?

Reply:
new "success" mentions you're running in the background